__tests__/integration/transaction.test.js

```diff
@@ -50,4 +50,51 @@ describe('transaction routes', () => {
     expect(response.body).toHaveProperty('version', tx.version);
     done();
   });
+
+  it('test confirmation blocks', async done => {
+    // Generates a transaction
+    const tx = await wallet1.injectFunds(1);
+    const txId = tx.hash;
+
+    // Queries for this transaction
+    const response = await TestUtils.request
+      .get('/wallet/tx-confirmation-blocks')
+      .query({
+        id: txId
+      })
+      .set({ 'x-wallet-id': wallet1.walletId });
+
+    expect(response.status).toBe(200);
+
+    // We might have some blocks while we are getting these requests
+    // so it's not easy to do an assert here
+    const height = await TestUtils.getFullNodeNetworkHeight();
+    const txData1 = await TestUtils.getFullNodeTransactionData(txId);
+
+    let firstBlockHeight = txData1.meta.first_block_height;
+    if (firstBlockHeight === null) {
+      // With this await here we know that we have at least one block confirming it
+      firstBlockHeight = Math.min(await TestUtils.waitNewBlock(height), height + 1);
```

Comment:
We should request the `txData` again instead of assuming the value of `firstBlockHeight`

Reply:
Makes sense, done here https://github.com/HathorNetwork/hathor-wallet-headless/pull/231/commits/50e018c12986087b387eaf5d5d48fd51d1749046